buildSrc/src/main/kotlin/terasology-module.gradle.kts

```diff
@@ -16,89 +16,69 @@
+// Complicating things more the use of publish.gradle to centralize logic means modules and engine bits are treated the same, yet we need to vary modules
+// Temporary workaround: default modules to bypass release management: master branch builds will still make snapshot builds for the snapshot repo
+// If a module actually wants release management simply include `"isReleaseManaged" : true` in module.txt - this is needed for the engine repo embedded modules
+// One option would be to slowly convert modulespace to default to a `develop` + `master` setup living in harmony with associated automation/github tweaks
+// Alternatively one more round of refactoring could switch to Git tags, a single `master` branch, and possible other things to help match snaps/PR builds somehow?
+val bypassModuleReleaseManagement by extra("true")
+
+val moduleDepends = mutableListOf<String>()
+val moduleFile = file("module.txt")
 
 // The module file should always exist if the module was correctly created or cloned using Gradle
 if (!moduleFile.exists()) {
     println("Y U NO EXIST MODULE.TXT!")
-    throw new GradleException("Failed to find module.txt for " + project.name)
+    throw GradleException("Failed to find module.txt for " + project.name)
 }
 
 //println "Scanning for dependencies in module.txt for " + project.name
-def slurper = new JsonSlurper()
-def moduleConfig = slurper.parseText(moduleFile.text)
-for (dependency in moduleConfig.dependencies) {
-    if (dependency.id != "engine") {
-        moduleDepends += dependency.id
+val slurper = JsonSlurper()  // 😂 using groovy's json parser in kotlin
+val moduleConfig: Map<String, Any> = slurper.parseText(moduleFile.readText())!! as Map<String, Any>
+for (dependency in moduleConfig["dependencies"] as List<Map<String, String>>) {
+    if (dependency["id"] != "engine") {
+        moduleDepends.add(dependency["id"]!!)
     }
 }
 
 // Gradle uses the magic version variable when creating the jar name (unless explicitly set somewhere else I guess)
-version = moduleConfig.version
-
-import org.gradle.internal.logging.text.StyledTextOutputFactory
-import static org.gradle.internal.logging.text.StyledTextOutput.Style
+version = moduleConfig["version"] as String
 
 // Check for an outright -SNAPSHOT in the loaded version - for ease of use we want to get rid of that everywhere, so warn about it and remove for the variable
-def undesiredSnapshotTag = version.endsWith("-SNAPSHOT")
+val undesiredSnapshotTag = version.toString().endsWith("-SNAPSHOT")
 if (undesiredSnapshotTag) {
     println("Taking off undesired -SNAPSHOT")
-    version -= "-SNAPSHOT"
-    def out = services.get(StyledTextOutputFactory).create("an-ouput")
-    out.withStyle(Style.FailureHeader).println("WARNING: Module " + project.name + " is explicitly versioned as a snapshot in module.txt, please remove '-SNAPSHOT'")
```

Comment:
Oh, so there is one feature I didn't preserve. The extra styling on the `please remove -SNAPSHOT` message. I was afraid it was using some groovy-specific formatter or something.

In hindsight, it looks like it's probably not groovy-specific? So I guess I probably could put it back.  
On the other hand, it is importing from an `.internal` package.